src/lib/inference/serving/spark-runtime-receipt.ts

```
@@ -0,0 +1,581 @@
+    fs.writeFileSync(fd, `${JSON.stringify(receipt)}\n`, "utf8");
+    fs.fsyncSync(fd);
+    fs.closeSync(fd);
+    fd = undefined;
+    fs.renameSync(temporary, filePath);
+    fsyncDirectory(stateDir);
+  } finally {
+    if (fd !== undefined) fs.closeSync(fd);
+    try {
+      fs.unlinkSync(temporary);
+    } catch (error) {
+      if ((error as NodeJS.ErrnoException).code !== "ENOENT") throw error;
```

Biome (2.5.5)

[error] Unsafe usage of 'throw'.

(lint/correctness/noUnsafeFinally)